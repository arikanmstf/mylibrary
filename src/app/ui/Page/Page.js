/**
 * Web Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React from 'react';
import styled from 'styled-components';
import type { Node } from 'react';
import defaultStyle from './style.json';
import type { PageProps } from './types';

const PageContainer = styled.div`
position: absolute;
top: 62px;
padding: 16px;
width: 100%;
height: auto;
box-sizing: border-box;
`;

const Page = (props: PageProps): ?Node => {
  const { style, ...rest } = props;
  const mergedStyle = {
    ...defaultStyle,
    ...style,
  };

  return <PageContainer style={mergedStyle} {...rest} />;
};

export default Page;
