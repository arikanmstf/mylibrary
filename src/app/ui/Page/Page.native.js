/**
 * Native Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React from 'react';

import type { Node } from 'react';
import { Container } from 'native-base';
import type { PageProps } from './types';
import defaultStyle from './style.json';

const Page = (props: PageProps): ?Node => {
  const { style, ...rest } = props;
  const mergedStyle = {
    ...defaultStyle,
    ...style,
  };

  return <Container style={mergedStyle} {...rest} />;
};

export default Page;
