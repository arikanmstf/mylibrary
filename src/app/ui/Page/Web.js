/**
 * Web Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import * as React from 'react';
import styled from 'styled-components';
import defaultStyle from './style.json';
import type { PageProps } from './types';

const PageContainer = styled.div`
position: absolute;
top: 62px;
padding: 16px;
width: 100%;
height: 100%;
box-sizing: border-box;
`;

const Page = (props: PageProps): ?React.Node => {
  const { style, ...rest } = props;
  const mergedStyle = {
    ...defaultStyle,
    ...style,
  };

  return <PageContainer style={mergedStyle} {...rest} />;
};

export default Page;
