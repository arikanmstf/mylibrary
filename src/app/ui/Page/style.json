{
  "backgroundColor": "#fffff0",
  "padding": 16
}
