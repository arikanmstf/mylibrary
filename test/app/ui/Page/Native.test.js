/**
 * Native Component Test Template By => create-module script
 * @version 1.0.0
 *
 */

import React from 'react';
import { shallow } from 'enzyme';
import Page from 'ui/Page/Native';

describe('test/app/ui/Page/Native.test.js', () => {
  it('Render', () => {
    const props = {};
    const wrapper = shallow(<Page {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
