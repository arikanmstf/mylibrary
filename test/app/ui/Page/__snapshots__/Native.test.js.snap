// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/ui/Page/Native.test.js Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "style": Object {
        "backgroundColor": "#fffff0",
        "padding": 16,
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "style": Object {
          "backgroundColor": "#fffff0",
          "padding": 16,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
